src/interfaces/IBPool.sol

```diff
@@ -54,53 +131,141 @@ interface IBPool is IERC20 {
     uint256 maxPrice
   ) external returns (uint256 tokenAmountIn, uint256 spotPriceAfter);
 
+  /**
+   * @notice Joins a pool providing only a single token, and a specific amount of tokenIn
+   * @param tokenIn The address of the token to swap in and join
+   * @param tokenAmountIn The amount of token to join
+   * @param minPoolAmountOut The minimum amount of pool token to receive
+   * @return poolAmountOut The amount of pool token received
+   */
   function joinswapExternAmountIn(
     address tokenIn,
     uint256 tokenAmountIn,
     uint256 minPoolAmountOut
   ) external returns (uint256 poolAmountOut);
 
+  /**
+   * @notice Joins a pool expecting a specific amount of pool token, and providing only a single token
+   * @param tokenIn The address of the token to swap in and join
+   * @param poolAmountOut The amount of pool token to receive
+   * @param maxAmountIn The maximum amount of token to introduce to the pool
+   * @return tokenAmountIn The amount of token in introduced
+   */
   function joinswapPoolAmountOut(
```

Comment:
Joins a pool providing a single token in, specifying the exact amount of pool tokens received